src/frequenz/sdk/timeseries/grid.py

```diff
@@ -194,33 +194,29 @@ def initialize(
 
     fuse: Fuse | None = None
 
-    if grid_connections_count == 0:
-        fuse = Fuse(max_current=Current.zero())
-        _logger.info(
-            "No grid connection found for this microgrid. This is normal for an islanded microgrid."
-        )
-    elif grid_connections_count > 1:
-        raise RuntimeError(
-            f"Expected at most one grid connection, got {grid_connections_count}"
-        )
-    else:
-        if grid_connections[0].metadata is None:
-            raise RuntimeError("Grid metadata is None")
-
-        # The current implementation of the Component Graph fails to
-        # effectively convert components from a dictionary representation to
-        # the expected Component object.
-        # Specifically for the component metadata, it hands back a dictionary
-        # instead of the expected ComponentMetadata type.
-        metadata = grid_connections[0].metadata
-        if isinstance(metadata, dict):
-            if fuse_dict := metadata.get("fuse", None):
-                fuse = Fuse(
-                    max_current=Current.from_amperes(fuse_dict.get("max_current", 0.0))
+    match grid_connections_count:
+        case 0:
+            fuse = Fuse(max_current=Current.zero())
+            _logger.info(
+                "No grid connection found for this microgrid. "
+                "This is normal for an islanded microgrid."
+            )
+        case 1:
+            metadata = grid_connections[0].metadata
+            if metadata is not None and metadata.fuse is not None:
+                fuse = Fuse(max_current=Current.from_amperes(metadata.fuse.max_current))
+
+            if metadata is None:
+                _logger.warning(
+                    "Unable to get grid metadata, the grid connection point is "
+                    "considered to have no fuse"
                 )
```

Comment:
I think we could avoid checking for metadata and fuse conditions twice as follows:
```python
            if metadata is None:
                _logger.warning(
                    "Unable to get grid metadata, the grid connection point is "
                    "considered to have no fuse"
                )
            elif metadata.fuse is None:
                _logger.warning("The grid connection point does not have a fuse")
            else:
                fuse = Fuse(max_current=Current.from_amperes(metadata.fuse.max_current))
```